EIPS/eip-5187.md

```diff
@@ -0,0 +1,102 @@
+category: ERC
+created: 2022-04-17
+requires: 721, 1155
+---
+
+## Abstract
+This standard is an extension of [EIP-1155](./eip-1155.md). It proposes introducing the concept of independent, multiple, and leasable rights of use to enable NFT to be leased out for different cycles while ownership remains with the owner.
+
+## Motivation
+The traditional [EIP-721](./eip-721.md) and EIP-1155 focus more on ownership. However, NFTs as digital assets are more prominent in use than ownership. Taking artistic NFTs as an example, NFT artists may wish to rent out the use rights of their works to media companies for an allotted time, or NFT musicians may wish to make their music available to listeners as per playing duration. 
+Therefore, to better serve NFT developers to meet such needs and develop more sophisticated NFT products, we propose directly introducing rentable usage rights to complement the ERC standard.
+
+
+## Specification
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+```solidity
+pragma solidity ^0.8.0;
+
+interface IRental is IERC1155 {
+    /**
+     * @notice This emits when user rent NFT
+     * @param id The id of the current token
+     * @param user The address to rent the NFT usage rights
+     * @param amount The amount of usage rights
+     * @param expires The specified period of time to rent
+     **/
+    event Rented(
+        uint256 indexed id,
+        address indexed user,
+        uint256 amount,
+        uint256 expires
+    );
+
+    /**
+     * @notice This emits when the NFT owner takes back the usage rights from the tenant (the `user`) 
+     * @param id The id of the current token
+     * @param user The address to rent the NFT's usage rights
+     * @param amount Amount of usage rights
+     **/
+    event TakeBack(uint256 indexed id, address indexed user, uint256 amount);
+
+    /**
+     * @notice Function to rent out usage rights
+     * @param from The address to approve
+     * @param to The address to rent the NFT usage rights
+     * @param id The id of the current token
+     * @param amount The amount of usage rights
+     * @param expires The specified period of time to rent
+     **/
+    function safeRent(
+        address from,
+        address to,
+        uint256 id,
+        uint256 amount,
+        uint256 expires
+    ) external;
+
+    /**
+     * @notice Function to take back usage rights after the end of the tenancy
+     * @param user The address to rent the NFT's usage rights
+     * @param tokenId The id of the current token
+     * @param amount The amount of usage rights
+     **/
+    function takeBack(
+        address user,
+        uint256 tokenId,
+        uint256 amount
+    ) external;
+
+    /**
+    * @notice Return the NFT to the NFT owner’s address
+    **/
+    function ownerOf(uint256 id) external view returns (address);
```

Comment:
There may be multiple owners.

Reply:
The ownership of NFT belongs to only one address and other permissions are attached to the ownership. 

(The usage rights involved in this proposal can be split and distributed to multiple address)

Reply:
Not sure what I was thinking... 🤦

Reply:
EIP-1155 tokens can have multiple owners.

Reply:
EIP-1155 allows multiple token holders, but "holding" a token does not necessarily mean that the holder really "owns" that EIP-1155 token, especially in a lease-enabled scenario where the holder may just be the lessee of the token, not the lessor/owner of the token. 

Therefore, we need to distinguish the concepts of "token holder" and "token owner", and ownership only belongs to a real "owner" address. So we use the updated "propertyRightOf" function to identify the true "owner" who owns the "property right" of the token.

Reply:
Not sure if I understand fully, but I'll assume you know what you're talking about for now.